### PR TITLE
Use personal access token when checking out repo in `cut_release_branch` workflow

### PR DESCRIPTION
**Pull request by msasikanth**
Since we want the release branch creation to run a different workflow (triggering demo release). We need to use PAT (Personal Access Token) instead of GITHUB_TOKEN in GH Actions to do that.

### CODE REVIEW

File: .github/workflows/cut_release_branch.yml
Comment (Honey14) [inline .github/workflows/cut_release_branch.yml:13]: Is the default token `GITHUB_TOKEN`, if we don't mention PAT?

**Review by Honey14 — APPROVED** [review]
LGTM, have to test this once merged

File: .github/workflows/cut_release_branch.yml
Comment (msasikanth) [reply .github/workflows/cut_release_branch.yml:13]: Yep. You can look at the checkout default over here
https://github.com/actions/checkout

```
    # Personal access token (PAT) used to fetch the repository. The PAT is configured
    # with the local git config, which enables your scripts to run authenticated git
    # commands. The post-job step removes the PAT.
    #
    # We recommend using a service account with the least permissions necessary. Also
    # when generating a new PAT, select the least scopes necessary.
    #
    # [Learn more about creating and using encrypted secrets](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/creating-and-using-encrypted-secrets)
    #
    # Default: ${{ github.token }}
    token: ''
```

**Pull request merged**
